### PR TITLE
Minor improvements to DNS proxy around `notifyOnDNSMsg()`

### PR DESCRIPTION
**Pull request by christarazi**
- dnsproxy: Fix misordered DNS msg callback
- daemon, fqdn: Defer logging DNS reply / request

### CODE REVIEW

**Review by jibi — APPROVED** [review]

File: daemon/cmd/fqdn.go [hunk 1]
```diff
@@ -522,7 +522,7 @@ func (d *Daemon) notifyOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, epI
 			AnswerTypes:       recordTypes,
 		}),
 	)
-	record.Log()
+	defer record.Log()
```
Comment (jrajahalme) [inline daemon/cmd/fqdn.go]: nit: Could also move the record formation and logging to the end to this function, as there are no `return`s in between?

**Review by jrajahalme — APPROVED** [review]
One small nit.

**Comment by jrajahalme** [comment]
@christarazi Given that there was a master breakage in waiting for DNS responses to be plumbed, it is likely that the DNS proxy test fails are not due to this PR, but it would be prudent to rebase and retest to be sure.

**Comment by christarazi** [comment]
Most of CI has passed, except for Travis master breakage and net-next failed to provision. Rebasing to re-run those two.

**Comment by christarazi** [comment]
/test-1.26-net-next

**Pull request merged**